batch_tasks/resource/hook/batch_tasks.py

```diff
@@ -0,0 +1,385 @@
+        the entity id.
+        If the entity is a hierarchical you will always get the entity
+        type TypedContext, once retrieved through a get operation you
+        will have the "real" entity type ie. example Shot, Sequence
+        or Asset Build.
+
+        *source* dictionary containing information about the source of the
+        event, application id, current user etc.
+
+        *values* is a dictionary containing potential user settings
+
+        *event* the unmodified original event
+        '''
+
+        # Only discover the action if any selection is made.
+        if entities:
+            return True
+
+        return False
+
+    def _translate_event(self, session, event):
+        '''Return *event* translated structure to be used with the API.'''
+
+        _uid = event['source']['id']
+        _source = event['source']
+
+        _values = event['data'].get('values', {})
+        _selection = event['data'].get('selection', [])
+
+        _entities = list()
+        for entity in _selection:
+            _entities.append(
+                (
+                    self._get_entity_type(entity), entity.get('entityId')
+                )
+            )
+
+        return [
+            _uid,
+            _entities,
+            _source,
+            _values,
+            event
+        ]
+
+    def _get_entity_type(self, entity):
+        '''Return translated entity type tht can be used with API.'''
+        entity_type = entity.get('entityType')
+        object_typeid = None
+
+        for schema in self._session.schemas:
+            alias_for = schema.get('alias_for')
+
+            if (
+                alias_for and isinstance(alias_for, dict) and
+                alias_for['id'].lower() == entity_type and
+                object_typeid == alias_for.get(
+                    'classifiers', {}
+                ).get('object_typeid')
+            ):
+
+                return schema['id']
+
+        for schema in self._session.schemas:
+            alias_for = schema.get('alias_for')
+
+            if (
+                alias_for and isinstance(alias_for, basestring) and
+                alias_for.lower() == entity_type
+            ):
+                return schema['id']
+
+        for schema in self._session.schemas:
+            if schema['id'].lower() == entity_type:
+                    return schema['id']
+
+        raise ValueError(
+            'Unable to translate entity type.'
+        )
+
+    def _launch(self, event):
+        args = self._translate_event(
+            self._session, event
+        )
+
+        interface = self._interface(
+            self._session, *args
+        )
+
+        if interface:
+            return interface
+
+        response = self.launch(
+            self._session, *args
+        )
+
+        return self._handle_result(
+            self._session, response, *args
+        )
+
+    def launch(self, session, uid, entities, source, values, event):
```

Comment:
This method is doing a lot of work and over 100 lines long, it probably should be broken down into smaller methods, which are named more appropriately.